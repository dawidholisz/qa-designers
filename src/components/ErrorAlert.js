import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

const ErrorAlert = ({ isVisible, header, errorMessage, refreshWeather, closeModal }) => {
  return (
    <div className={classNames('modal', { '-visible': isVisible })}>
      <div className="modal__dialog">
        <button className="close-button" onClick={closeModal} data-testid="close-button">
          <FontAwesomeIcon icon="times" size="lg" />
        </button>

        <h1>{header}</h1>

        sf
        <div className="modal__description">
          <p>{errorMessage}</p>
        </div>

        <button
          className="set-button"
          data-testid="refresh-button"
          onClick={() => {
            refreshWeather()
          }}
        >
          Refresh
        </button>
      </div>
    </div>
  )
}

ErrorAlert.propTypes = {
  errorMessage: PropTypes.string.isRequired,
  header: PropTypes.string.isRequired,
  isVisible: PropTypes.bool.isRequired,
  refreshWeather: PropTypes.func.isRequired,
  closeModal: PropTypes.func.isRequired,
}
export default ErrorAlert
