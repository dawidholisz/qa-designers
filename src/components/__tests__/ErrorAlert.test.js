import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import ErrorAlert from '../ErrorAlert'

const renderComponent = props => {
  return render(<ErrorAlert {...props} />)
}
const props = {
  header: 'header',
  errorMessage: 'Error Message',
  isVisible: true,
  closeModal: jest.fn(),
  refreshWeather: jest.fn(),
}

describe('<ErrorAlert/>', () => {
  it('render Error alert', () => {
    const { container } = renderComponent(props)
    expect(container.firstChild).toMatchSnapshot()
  })
  it('close Error alert', () => {
    const { getByTestId } = renderComponent(props)
    fireEvent.click(getByTestId('close-button'))
    expect(props.closeModal).toHaveBeenCalled()
  })
  it('refresh weather', () => {
    const { getByTestId } = renderComponent(props)
    fireEvent.click(getByTestId('refresh-button'))
    expect(props.refreshWeather).toHaveBeenCalled()
  })
})
