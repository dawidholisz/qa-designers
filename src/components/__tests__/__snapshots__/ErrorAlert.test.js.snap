// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorAlert/> render Error alert 1`] = `
<div
  class="modal -visible"
>
  <div
    class="modal__dialog"
  >
    <button
      class="close-button"
      data-testid="close-button"
    />
    <h1>
      header
    </h1>
    <div
      class="modal__description"
    >
      <p>
        Error Message
      </p>
    </div>
    <button
      class="set-button"
      data-testid="refresh-button"
    >
      Refresh
    </button>
  </div>
</div>
`;
